Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py

```diff
@@ -735,7 +735,7 @@ def cleanup_records(signal: Optional[Dict[str, Any]]) -> Optional[Dict[str, Any]
     for rec in signal['allRecords']:
         field_names = list(rec.keys())
         for field_name in field_names:
-            if (field_name.startswith('bro') and len(rec.get(field_name)) == 0):
```

Comment:
Could you please explain this change, is it a bug fix? 
Does the record always exist? Is the record always a list? 
Is the purpose of this change to remove the None or empty list values?

Reply:
Yes, this is a bug fix. To give you a context, one of our Sumo Logic customers reported the below error:

```
Raw Response section:
Failed to execute test-module command.
Error:
object of type 'int' has no len()
```

So from the error, the field's value could also have an integer value. The `len()` works only on container-type data structures (string, list, object, or tuple). Hence, checking the length of the integer type will result in such an error.

I have updated the code to remove the field only if it has an empty data structure value ([], (), "", {}).

Reply:
Hey @milan-sumo, 
Please notice that after your change None values won't be removed.
Does the record always exist? Is the record always a data structure?
Why there is an int there? Please explain.

Reply:
1. We want to remove the field only if its value is an empty data structure, the same as the existing behavior. 
2. We don't know if the value of the field will always exist and will always be a data structure. From the error reported by the customer, it seems that the value can be of primitive type (int, boolean, etc) as well. 

Here is the complete error given by the customer:
```
==
Raw Response section:
Failed to execute test-module command.
Error:
object of type 'int' has no len()
==

  *   Error reported in fetch history is also given below:
==
Error: Script failed to run:
Error: [Traceback (most recent call last):

   File "<string>", line 1144, in <module>
   File "<string>", line 1137, in main
   File "<string>", line 7499, in return_error
Exception: Failed to execute fetch-incidents command.
Error:
object of type 'int' has no len()
] (2604) (2603)
===
```

So from the error, we found that the issue occurred in the `fetch_incidents()` function. Inside the `fetch_incidents()` function, there are a few places where the `len()` function is being called. Out of all, this is the only place where the error can occur if the value is not of type list, tuple, object, or string.

Hence, I have just changed the way the code checks for the empty data structures. Instead of using the `len()` function over the value, the value is directly being compared with the empty data structures. This way it won't call the `len()` function when the field has a value other than the data structure type, and won't result in the error `object of type 'int' has no len()`. The behavior of the code has remained the same as earlier.